providers/apache/impala/src/airflow/providers/apache/impala/hooks/impala.py

```diff
@@ -45,3 +46,30 @@ def get_conn(self) -> Connection:
             database=connection.schema,
             **connection.extra_dejson,
         )
+
+    @property
+    def sqlalchemy_url(self) -> URL:
+        """Return a `sqlalchemy.engine.URL` object constructed from the connection."""
+        conn = self.get_connection(self.get_conn_id())
+        extra = conn.extra_dejson or {}
+
+        required_attrs = ["host", "login"]
+        for attr in required_attrs:
+            if getattr(conn, attr) is None:
+                raise ValueError(f"Impala Connection Error: '{attr}' is missing in the connection")
+
+        query = {k: str(v) for k, v in extra.items() if v is not None and k not in ["__extra__"]}
```

Comment:
```suggestion
        query = {k: str(v) for k, v in extra.items() if v is not None and k != "__extra__"}
```

Cool! Thanks! Didn't know we have this. Then maybe we can try something like this